Face/Identify/Program.cs

```diff
@@ -0,0 +1,218 @@
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Create a person group person.
+                    Person p = new Person();
+                    p.Name = targetImageFileName;
+                    p.UserData = "Person for sample";
+                    Console.WriteLine($"Create a person group person '{p.Name}'");
+                    p.PersonId = (await Client.LargePersonGroupPerson.CreateAsync(GroupId, p.Name)).PersonId;
+
+                    // Add face to the large person group.
+                    Console.WriteLine($"Add faces to the large person group from image `{targetImageFileName}`.{Environment.NewLine}");
+                    PersistedFace faces = await Client.LargePersonGroupPerson.AddFaceFromStreamAsync(GroupId, p.PersonId, stream, targetImageFileName);
+
+                    if (faces == null)
+                    {
+                        Console.WriteLine($"[Error] No persisted face from image `{targetImageFileName}`.");
+                        return;
+                    }
+                }
+            }
+
+            // Start to train the large person group.
+            await Client.LargePersonGroup.TrainAsync(GroupId);
+
+            // Wait until the training is completed.
+            while (true)
+            {
+                await Task.Delay(1000);
+                var status = await Client.LargePersonGroup.GetTrainingStatusAsync(GroupId);
+                Console.WriteLine($"Response: Success. Group \"{GroupId}' training process is {status.Status}");
+
+                if (status.Status != TrainingStatusType.Running)
+                {
+                    break;
+                }
+            }
+
+            List<Guid> faceIds = new List<Guid>();
+            // Read image file. 
+            using (FileStream stream = new FileStream(Path.Combine("Images", SourceImageFileName), FileMode.Open))
+            {
+                // Detect faces from image stream. 
+                IList<DetectedFace> detectedFaces = await Client.Face.DetectWithStreamAsync(stream);
+                if (detectedFaces == null || detectedFaces.Count == 0)
+                {
+                    Console.WriteLine($"[Error] No face detected from image `{SourceImageFileName}`.");
+                    return;
+                }
+
+                Console.WriteLine($"{detectedFaces.Count} faces detected from image `{SourceImageFileName}`.");
+                if (detectedFaces[0].FaceId == null)
+                {
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for Identify purpose.");
+                    return;
+                }
+
+                // Add detected faceId to faceIds.
+                foreach (var face in detectedFaces)
+                {
+                    faceIds.Add(face.FaceId.Value);
```

Comment:
`FaceId` can be null.

foreach (var face in detectedFaces)
                {
                    if (face.FaceId != null)
                    {
                        faceIds.Add(face.FaceId.Value);
                    }
                }

Reply:
done